src/cluster/mcl.h

```diff
@@ -64,29 +65,45 @@ class MCL: public ClusteringAlgorithm {
 	atomic_ullong dense_list_time = {0};
 #endif
 public:
+	~MCL(){};
 	void run();
-	string get_key();
 	string get_description();
+	static string get_key(){
+		return "mcl";
+	}
 };
 
 template <typename T>
 class SparseMatrixStream : public Consumer {
 	size_t n;
 	vector<Eigen::Triplet<T>> data;
 	LazyDisjointSet<uint32_t>* disjointSet;
+	mutex mutex;
```

Comment:
I don't think you need a mutex here. This should not be called concurrently.

Reply:
Thanks. I was not sure.